site/src/components/NewsletterSignup.js

```diff
@@ -1,20 +1,67 @@
-import React from 'react'
-import MailchimpForm from 'react-mailchimp-subscribe'
+import React, { Component } from 'react'
+import jsonp from 'jsonp'
 import './NewsletterSignup.scss'
 
-export default function NewsletterSignup() {
-  return (
-    <MailchimpForm
-      className="NewsletterSignup"
-      action="https://lastcallmedia.us9.list-manage.com/subscribe/post?u=d60cafbefdea2f1ee497ac747&amp;id=3dcf6c9dcc"
-      messages={{
-        inputPlaceholder: 'Subscribe for updates',
-        btnLabel: 'Subscribe',
-        sending: 'Sending...',
-        success:
-          'Subscribed! Please check your e-mail and follow the confirmation link.',
-        error: "Oops, we weren't able to subscribe you!",
-      }}
-    />
-  )
-}
+const action = 'https://lastcallmedia.us9.list-manage.com/subscribe/post?u=d60cafbefdea2f1ee497ac747&amp;id=3dcf6c9dcc'
+const getAjaxUrl = url => url.replace('/post?', '/post-json?')
+
+export default class NewsletterSignup extends Component {
+  constructor(props, ...args) {
+    super(props, ...args)
+    this.state = {
+      status: null
+    }
+  }
+  onSubmit = e => {
+    e.preventDefault()
+    if (!this.input.value || this.input.value.length < 5 || this.input.value.indexOf('@') === -1) {
+      this.setState({
+        status: 'error'
+      })
+      return
+    }
+    const url = getAjaxUrl(action) + `&EMAIL=${encodeURIComponent(this.input.value)}`;
+    this.setState(
+      {
+        status: 'sending'
+      }, () => jsonp(url, {
+        param: 'c'
+      }, (err, data) => {
+        if (err) {
+          this.setState({status: 'error'})
+        } else if (data.result !== 'success') {
+          this.setState({status: 'error'})
+        } else {
+          this.setState({status: 'success'})
+        }
+      })
+    )
+  }
+  render() {
+    const { status } = this.state
+    return (
+      <form className={['NewsletterSignup', status].join(' ')} action={action} method="post" noValidate>
+        <label htmlFor="NewsletterSignup--text-field">Subscribe for updates</label>
+        <div>
+          <input
+            id="NewsletterSignup--text-field"
+            ref={node => (this.input = node)}
+            type="email"
+            defaultValue=""
+            name="EMAIL"
+            required={true}
+            placeholder='Email'
+          />
+          <button
+            disabled={this.state.status === "sending" || this.state.status === "success"}
+            onClick={this.onSubmit}
+            type="submit"
```

Comment:
As long as we’re customizing this form, let’s make it accessible. Empty buttons and placeholder only inputs are no good, right?

Reply:
Oops - just spotted the label. So I guess the button is the only question

Reply:
Made the form more accessible, but I'm not sure how best to handle Mailchimp's error messages. For instance, if I try to subscribe with invalid email adress of "test@face" Mailchimp responds with `The domain portion of the email address is invalid (the portion after the @: face)`. We don't want to display that to user, it's oddly worded and rather verbose.

Reply:
Cool. Looks good to me. Also, just a stylistic thing, but since were using es6, you could use string interpolation to do the form class:

‘’’js
‘NewsletterSignup ${status}’
‘’’
This looks fine as-is though. Thanks!

Reply:
Argh, on my phone. Backticks trigger string interpolation

Reply:
Oh nice, I'll switch to this. I was really bothered that a `join` was the cleanest method I could find to put those two classes together.

Reply:
Sounds good. You’ll want to start a new branch for that off master. I squash merged this. Feel free to merge it yourself when it’s ready.

Reply:
Actually, `"`NewsletterSignup ${status}`"` didn't work for me inside an attribute value. I think this is a React thing. I'll just leave it as is.